### PR TITLE
New class for processing modules

### PR DESCRIPTION
**Pull request by onosek**

### CODE REVIEW

File: productmd/modules.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (C) 2015  Red Hat, Inc.
```
Comment (lubomir) [inline productmd/modules.py]: Update the year?

File: productmd/modules.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        if not koji_tag:
+            raise ValueError("Non-empty 'koji_tag' is expected")
+
+        for param_name, param in {"variant": variant, "koji_tag": koji_tag, "modulemd_path": modulemd_path}.items():
+            if not param:
+                raise ValueError("Non-empty '%s' is expected" % param_name)
+
+        if type(rpms) not in (list, tuple):
+            raise ValueError("Wrong type of 'rpms'")
+        if not len(rpms):
```
Comment (lubomir) [inline productmd/modules.py]: This should be just `not rpms`. No need to calculate lenght.

File: tests/test_modules.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+
+# Copyright (C) 2015  Red Hat, Inc.
```
Comment (lubomir) [inline tests/test_modules.py]: Update year.

**Comment by lubomir** [comment]
It would probably be also nice to add a `modules` property to `productmd.compose.Compose` class for easy access.

File: productmd/modules.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+    def __getitem__(self, variant):
+        return self.modules[variant]
+
+    def __delitem__(self, variant):
+        del self.modules[variant]
+
+    @staticmethod
+    def parse_uid(uid):
+        if not isinstance(uid, str):
```
Comment (lubomir) [inline productmd/modules.py]: This should use `six.string_types`.

File: productmd/modules.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+        if uid_dict["version"] is None:
+            uid_dict["version"] = ""
+        if uid_dict["context"] is None:
+            uid_dict["context"] = ""
+
+        return uid_dict
+
+    def _check_uid(self, uid):
+        if not isinstance(uid, str):
```
Comment (lubomir) [inline productmd/modules.py]: This should use `six.string_types`.

File: productmd/modules.py [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        uid += ":%s" % uid_dict['context'] if uid_dict['context'] else ""
+        return uid, uid_dict
+
+    def serialize(self, parser):
+        self.validate()
+        data = parser
+        self.header.serialize(data)
+        data["payload"] = {}
+        self.compose.serialize(data["payload"])
+        data["payload"]["modules"] = {}
```
Comment (lubomir) [inline productmd/modules.py]: This line can be removed.

Comment (onosek) [reply productmd/modules.py]: done

File: productmd/modules.py [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+        metadata["metadata"] = {
+            "uid": uid,
+            "name": name,
+            "stream": stream,
+            "version": version,
+            "context": context,
+            "koji_tag": koji_tag,
+        }
+        metadata["modulemd_path"] = modulemd_path
+        metadata["rpms"] = list(rpms)
```
Comment (lubomir) [inline productmd/modules.py]: If there are binary and debuginfo packages, they will all contribute to the same module. The lists here should be joined if there already are some rpms.

**Pull request merged**